packages/text-sets/src/loadTextSets.ts

```diff
@@ -89,5 +92,5 @@ export default async function loadTextSets() {
     })
   );
 
-  return Object.fromEntries(results);
+  return Object.fromEntries(results) as TextSets;
```

Comment:
`as` is a rather crude tool that should only be applied if no better tool exists, as it will actually ignore and suppress some potential errors. However, moving it to a typed variable or just setting the return type won't do the trick, because `fromEntries` doesn't play nice (typed as returning `any`).

This required updating the type system a bit as:

```ts
export enum TextSetType {
  Cover = 'cover',
  Step = 'step',
  SectionHeader = 'section_header',
  Editorial = 'editorial',
  Contact = 'contact',
  Table = 'table',
  List = 'list',
  Quote = 'quote',
}

export type TextSets = Partial<Record<TextSetType, TextSet[]>>;
```

And then `loadTextSets` becomes:

```ts
export default async function loadTextSets(): Promise<TextSets> {
  const results: [TextSetType, TextSet[]][] = await Promise.all(
    Object.values(TextSetType).map(async (name) => [
      name,
      await loadTextSet(name),
    ])
  );
  return Object.fromEntries(results);
}
```

The `Partial<>` is a bit annoying but I don't see a clean way around it? I feel this is cleaner, but it's probably just a nit really.

Reply:
If I am honest, I don't find your code easier to read or understand. I have pushed up my own version of the code, that is a little easier to follow. See https://github.com/GoogleForCreators/web-stories-wp/pull/12274/commits/ed4be89718d16dba892908a2f1f6afb18539bc63

Reply:
@barklund Can you take a look ?